Cloudinary-pwa-kit/packages/pwa-cloudinary/app/components/cloudinary-widgets/index.jsx

```diff
@@ -4,22 +4,26 @@ import { cloudinary } from '../../../config/default'
 
 const RenderCloudinaryGalleryWidget = ({ cloudinaryImageGallery = {} }) => {
 
-    useEffect(() => {
-        if (cloudinaryImageGallery && cloudinaryImageGallery.galleryEnabled) {
-            const gallery = document.querySelector('#cld-gallery')
-            const galleryOptions = cloudinaryImageGallery.cloudinaryImage.galleryWidget.options
-            galleryOptions.queryParam = cloudinary.CLD_PGW_TRACKING_PARAM
-            if (typeof window !== 'undefined' && window.cloudinary && window.cloudinary.galleryWidget) {
-                if (window.cldGallery) {
-                    gallery.style.visibility = 'hidden'
-                    cldGallery.update(galleryOptions)
-                    gallery.style.visibility = 'visible'
-                } else {
-                    window.cldGallery = window.cloudinary.galleryWidget(galleryOptions)
-                }
+    const renderGallery = async () => {
+        const gallery = document.querySelector('#cld-gallery')
+        const galleryOptions = cloudinaryImageGallery.cloudinaryImage.galleryWidget.options
+        galleryOptions.queryParam = cloudinary.CLD_PGW_TRACKING_PARAM
+        if (typeof window !== 'undefined' && window.cloudinary && window.cloudinary.galleryWidget) {
+            if (window.cldGallery) {
+                gallery.style.visibility = 'hidden'
```

Comment:
@yuval-cloudinary, I just tested, if we remove this `gallery.style.visibility = 'hidden'` and `gallery.style.visibility = 'visible'` it works fine, after our changes of `await cldGallery.update(galleryOptions)`

Reply:
do you want me to remove the visibility attribute or stick with it?

Reply:
@asad-rafter yes, please remove the visibility attribute. That would be the last change and I'll close and merge it